templates/types/streaming/fastapi/app/api/routers/chat.py

```diff
@@ -42,10 +40,11 @@ async def chat(
         chat_engine = get_chat_engine(
             filters=filters, params=params, event_handlers=[event_handler]
         )
-        response = await chat_engine.astream_chat(last_message_content, messages)
```

Comment:
waiting chat response here will block the whole VercelStreamResponse.

Reply:
we had two places that using the response result: `process_response_nodes` and `VercelStreamResponse` so we would need two `await` for the response. So there'd be two solution:
1. Create a new asyncio task for the response, then i think we can await the task result in both two places.
2. move the `process_response_nodes` to VercelStreamResponse - this is what i did.